flake.nix

```diff
@@ -60,25 +68,136 @@
           '';
         };
 
+        ipfs-sniffer = pkgs.buildGoModule rec {
+          pname = "ipfs-sniffer";
+          version = "master";
+          src = pkgs.fetchFromGitHub {
+            owner = "ipfs-search";
+            repo = pname;
+            rev = "4a7c4441fc9039ce96edffc6ab30d08ac4c2a5f6";
+            sha256 = "sha256-HO21gjQLqzunkbJcZ1Hs+AnaaXJbsiL5BET0MDDWJZ0=";
+          };
+          vendorSha256 = "sha256-xc1biJF4zicosSTFuUv82yvOYpbuY3h++rhvD+5aWNE=";
+        };
+
+        jaeger = pkgs.buildGoModule rec {
+          pname = "jaeger";
+          version = "jaegertracing";
+          src = pkgs.fetchFromGitHub {
+            owner = "jaegertracing";
+            repo = pname;
+            rev = "v1.25.0";
+            sha256 = "sha256-QzHWgjtCKtDVMNkVx82JqsslgIVuCso/xz6ZdQmmkNs=";
+          };
+          vendorSha256 = "sha256-f/DIAw8XWb1osfXAJ/ZKsB0sOmFnJincAQlfVHqElBE=";
+        };
+
+        kibana7-oss = prev.kibana7-oss.overrideAttrs (old: {
+          src = pkgs.fetchurl {
+            url = "https://artifacts.elastic.co/downloads/kibana/kibana-oss-7.8.1-linux-x86_64.tar.gz";
+            sha256 = "sha256-WWoOslKYWfoPc4wOU84QdxJln88JOmG8VhMaMtLraxs=";
+          };
+        });
+
+        elasticsearch7-oss = prev.elasticsearch7-oss.overrideAttrs (old: {
+          src = pkgs.fetchurl {
+            url = "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-7.8.1-linux-x86_64.tar.gz";
+            sha256 = "sha256-eJ7tt6daRd5EdWMQMYy8BBPnArsjB5t03fjqScKivcU=";
+          }; 
+        });
+
+        # using nodejs 14 despite upstream uses version 10 (EOL)
+        ipfs-search-api-server = 
+          let
+            repoSrc = pkgs.fetchFromGitHub {
+              owner = "ipfs-search";
+              repo = "ipfs-search-api";
+              rev = "8a6369d652263e574e026468d854284e1e2221fc";
+              sha256 = "sha256-kQ9Hczgb8CLgbBIlbzpLI5z9mwrVhplL5F9icGjrJKM=";
+            };
+          in
+            npmlock2nix."${final.hostPlatform.system}".build {
+              src = "${repoSrc}/server";
+              dontBuild = true;
+              installPhase = ''
+                mkdir -p $out/{bin,lib}
+
+                # copy npmlock2nix modules to lib
+                cp -r node_modules $out/lib/node_modules
+
+                # copy source files to lib
+                cp -r search $out/lib/search
+                cp -r metadata $out/lib/metadata
+                for file in esclient.js server.js types.js; do
+                  echo "#!$(${pkgs.which}/bin/which node)" > $out/lib/$file
```

Comment:
why don't you use pkgs.nodejs directly here ? (instead of pkgs.which)

Reply:
I thought this way will always work, no matter what npmlock2nix does with nodejs, or will do in the future. But we can change it to pkgs.nodejs if you prefer that.

Reply:
I see, you didn't want to specifically override the node version used, that makes sense.